frontend/src/components/HomePage.js

```diff
@@ -97,58 +95,52 @@ function DifficultySelection({ socket }) {
   };
 
   return (
-    <div className="container">
-      <MatchingDialog
-        initSeconds={30}
-        isOpen={isMatchingDialogOpen}
-        handleClose={stopFindingMatch}
-        matchStatus={matchStatus}
-        failedFindingMatch={failedFindingMatch}
-      />
-      <div className="row">
-        <div className="col-sm-12">
-          <form onSubmit={handleFormSubmit}>
-            <div className="radio">
-              <label>
-                <input
-                  type="radio"
-                  value="easy"
-                  checked={selectedDifficulty === "easy"}
-                  onChange={handleOptionChange}
-                />
-                Easy
-              </label>
-            </div>
-            <div className="radio">
-              <label>
-                <input
-                  type="radio"
-                  value="medium"
-                  checked={selectedDifficulty === "medium"}
-                  onChange={handleOptionChange}
-                />
-                Medium
-              </label>
-            </div>
-            <div className="radio">
-              <label>
-                <input
-                  type="radio"
-                  value="hard"
-                  checked={selectedDifficulty === "hard"}
-                  onChange={handleOptionChange}
-                />
-                Hard
-              </label>
-            </div>
-            <button className="btn btn-default" type="submit">
-              Match
-            </button>
-          </form>
-        </div>
-      </div>
-    </div>
+    <Grid container>
+      <Grid item xs={1} />
+      <Grid item xs={10}>
+        <Box display={"flex"} alignItems={"center"} justifyContent={"center"}>
+          <Typography variant={"h3"} ma={"2rem"}>
+            Random Matching
+          </Typography>
+        </Box>
+        <Box
+          display={"flex"}
+          flexDirection={"column"}
+          alignItems={"center"}
+          justifyContent={"center"}>
+          <Button
+            variant="contained"
+            onClick={handleDifficulty("easy")}
+            color={"secondary"}
+            sx={{ padding: "1rem 8rem 1rem 8rem", margin: "1rem" }}>
+            Easy
+          </Button>
+          <Button
+            variant="outlined"
+            onClick={handleDifficulty("medium")}
+            color={"secondary"}
+            sx={{ padding: "1rem 7rem 1rem 7rem", margin: "1rem" }}>
+            Medium
+          </Button>
+          <Button
+            variant="contained"
+            onClick={handleDifficulty("hard")}
+            color={"secondary"}
+            sx={{ padding: "1rem 8rem 1rem 8rem", margin: "1rem" }}>
+            Hard
+          </Button>
+          <MatchingDialog
+            initSeconds={30}
+            isOpen={isMatchingDialogOpen}
+            handleClose={stopFindingMatch}
+            matchStatus={matchStatus}
+            failedFindingMatch={failedFindingMatch}
+          />
```

Comment:
Could we abstract out the styling or perhaps create reusable components (e.g. buttons)? Some lines look repeated and Im wondering if we would reuse the styling in another page. 

May not be necessary if such button styling is specific to this page.

Reply:
Yup, I actually tried to have a css file however some of them not working when it is in css and having some weird result even they are given a same styling so would need a lot more time for that. For this pr the main purpose is to get every component there in every pages.

Reply:
Sure, we just have to keep in mind to improve in the future